plugins/inputs/linux_fscache/linux_fscache.go

```diff
@@ -0,0 +1,67 @@
+package linux_fscache
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type FSCache struct {
+	statFile string
+}
+
+func (f *FSCache) Description() string {
+	return "Get FS-Cache statistics from /proc/fs/fscache/stats"
+}
+
+func (f *FSCache) SampleConfig() string { return "" }
+
+func (f *FSCache) Gather(acc telegraf.Accumulator) error {
+
+	s, err := os.Stat(f.statFile)
+	if err != nil {
+		return err
+	} else if s.Size() == 0 {
+		return fmt.Errorf("fscache: %s zero length!", f.statFile)
+	}
```

Comment:
I think remove the stat + size check, if the file is size zero then just don't emit any metrics.  (Unless this is a very common error case but I don't think it is)

Reply:
done.